torch/onnx/_internal/exporter/_compat.py

```diff
@@ -153,12 +157,26 @@ def export_compat(
     else:
         args, kwargs = _get_torch_export_args(args, kwargs)
         if dynamic_shapes is None and dynamic_axes is not None:
-            dynamic_shapes = _from_dynamic_axes_to_dynamic_shapes(
-                model,
-                dynamic_axes=dynamic_axes,
-                input_names=input_names,
-                output_names=set(output_names or ()),
+            warnings.warn(
+                "# ⚠️ **dynamic_axes is not recommended to dynamo=True, "
+                "and might lead to torch._dynamo.exc.UserError: Constraints violated. "
+                "Please use dynamic_shapes for better user experience.** ⚠️",
```

Comment:
```suggestion
                "# 'dynamic_axes' is not recommended when dynamo=True, "
                "and may lead to 'torch._dynamo.exc.UserError: Constraints violated.' "
                "Supply the 'dynamic_shapes' argument instead if export is unsuccessful.",
```

Reply:
I thought some emoji making it more noticeable to users?

Reply:
True, but if not careful the messages will start competing with each other. I think it’s good if we reserve them for the status messages for now